buildSrc/src/main/java/com/google/firebase/gradle/plugins/GradleUtils.kt

```diff
@@ -0,0 +1,10 @@
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Copy
+
+fun Copy.fromDirectory(directory: Provider<File>) =
+    from(directory) {
+        into(directory.map { it.nameWithoutExtension })
+    }
```

Comment:
Not sure what this does, is it copying into the same directory? i.e. what if the directory has no extension?

also, not sure if you need nesting, instead of
```suggestion
fun Copy.fromDirectory(directory: Provider<File>) {
    from(directory)
    into(directory.map { it.nameWithoutExtension })
}
```

Reply:
This is different from 
```
from()
into()
```

If you do `from("daymon/myFolder")` it will copy everything under `myFolder`, not the folder itself.

The general solution to this is as such:

```
from("daymon/myFolder")
into("vlad/myFolder")
```

Where it copies all the contents of `myFolder` into a new `myFolder` under the directory you want. This isn't as straight forward when you have multiple directories that you need to do this for- in one `Copy` task.

This extension method fixes that.

```
    from(myFolder) {
        into(myFolder.nameWithoutExtension)
    }
```

Instead of _just_ copying the contents of the folder, this pastes them into a temp subdirectory (that doesn't exist beyond the task), and returns that temp folder instead. This way, you get the folder AND its contents, versus just the contents. Its a little work-around, abstracted by an extension method (`fromDirectory`).

Furthermore, `nameWithoutExtension` is not specific to files with extensions. This is supposed to be used on directories, which don't actually have extensions. `nameWithoutExtension` is an extension method on `File`- provided by `kotlin.io`. If the file doesn't have an extension, it just returns the file name. FWIW, `nameWithoutExtension` could be removed.

Reply:
Makes sense, thanks for a detailed explanation.